src/operator/batch_norm.cc

```diff
@@ -334,6 +339,30 @@ Operator *CreateOp<cpu>(BatchNormParam param, const int dtype, const TShape& sha
     }
   }
 #endif
+#if MXNET_USE_MKLDNN == 1
+  if (shape.ndim() == 4
+      && param.axis == mxnet::op::batchnorm::DEFAULT_AXIS
+      && shape[param.axis] % 8 == 0
+      && !mxnet::op::batchnorm::disable_mkl) {
+    switch (dtype) {
+      case mshadow::kFloat32:
+        op = new MKLDNNBatchNormOp<cpu, float>(param);
+        break;
+      default:
+        // MKL operator doesn't support half_t, so fall through
+        break;
+    }
+  }
+#define BATCHNORM_LOG_MKL_INFO() \
+  do { \
```

Comment:
why does this need a while loop?

Reply:
this is for macro definition,  so it expands/compiles correctly in all cases.